src/pages/suggestions/edit/[suggestion].tsx

```diff
@@ -0,0 +1,193 @@
+      return {
+        redirect: {
+          destination: "/403",
+          permanent: false,
+        },
+      }
+
+    return { props: { suggestion: data } }
+  } catch {
+    return { notFound: true }
+  }
+}
+
+interface EditSuggestionPageProps {
+  suggestion: Suggestion
+}
+
+const EditSuggestionPage = (props: EditSuggestionPageProps) => {
+  const user = useUser()
+  const supabase = useSupabaseClient<Database>()
+  const router = useRouter()
+  const dispatch = useDispatch()
+
+  const newSuggestion = useSelector((state: AppState) => state.editSuggestion)
+  const suggestion = useSelector((state: AppState) => state.editSuggestion.suggestion)
+  const activeArea = useSelector((state: AppState) => state.editSuggestion.activeArea)
+  const deletedInstruments = useSelector(
+    (state: AppState) => state.editSuggestion.deletedInstrumentUuids
+  )
+
+  //If the user visits a new edit page, clear the previously edited from redux
+  if (props.suggestion.id !== newSuggestion.lastEditedUuid) {
+    const suggestionInstruments: NewSuggestionInstrument[] = []
+    props.suggestion.song_instruments.forEach((element) => {
+      suggestionInstruments.push({
+        id: element.id,
+        description: element.description ?? "",
+        instrument: element.instrument,
+      })
+    })
+
+    dispatch(
+      updateEditSuggestion({
+        artist: props.suggestion.artist,
+        link: props.suggestion.link,
+        motivation: props.suggestion.motivation,
+        title: props.suggestion.title,
+        instruments: suggestionInstruments,
+      })
+    )
+
+    dispatch(updateLastDeletedInstrumentUuid([]))
+    dispatch(updateLastEditedUuid(props.suggestion.id))
+  }
+
+  const saveSuggestion = (onSuccess: () => void, onError: () => void) => {
+    if (user) {
+      updateSuggestion(supabase, suggestion, newSuggestion.lastEditedUuid ?? "")
+        .then((response) => {
+          if (response.error) {
+            onError()
+            return
+          }
+
+          const suggestionId = response.data.at(0)!.id
+          updateSuggestionInstruments(supabase, mapEditInstruments(suggestion, suggestionId))
+            .then((response) => {
+              if (response.error) {
+                onError()
+                return
+              }
+
+              deleteSuggestionInstruments(supabase, deletedInstruments)
+                .then((response) => {
+                  router.push(`/suggestions/${suggestionId}`).then(() => {
+                    onSuccess()
+                    dispatch(setActiveArea(Area.SongInformation))
+                    dispatch(updateLastEditedUuid(""))
+                    dispatch(updateLastDeletedInstrumentUuid([]))
+                  })
+                })
+                .catch(() => {
+                  onError()
+                })
+            })
+            .catch(() => {
+              onError()
+            })
+        })
+        .catch(() => {
+          onError()
+        })
```

Comment:
Putting it all in a try catch with await would have reduced the amount of catches here I think

Reply:
Changed to try/catch in new commit